x-pack/platform/plugins/shared/onechat/public/application/components/conversations/conversation_input/conversation_input_form.tsx

```diff
@@ -27,6 +29,13 @@ export const ConversationInputForm: React.FC<ConversationInputFormProps> = ({ on
   const [input, setInput] = useState('');
   const { sendMessage, pendingMessage } = useSendMessage();
   const { euiTheme } = useEuiTheme();
+  const { agents, isLoading, isFetched } = useOnechatAgents();
+  const agentId = useAgentId();
+  const hasActiveConversation = useHasActiveConversation();
+
+  // Find the current agent from the agents array
+  const currentAgent = agents?.find((agent) => agent.id === agentId);
+  const shouldDisableTextArea = !currentAgent && isFetched && !!agentId;
```

Comment:
3 conditions are:
- checks the current agent is still a valid agent
- checks that we have completed fetching all the agents
- checks that the current conversation actually has an agentId - this won't be true when users are on the landing page I.e. it should never be disabled in this instance

There might be easier ways to confirm this?